src/hotspot/cpu/riscv/riscv.ad

```diff
@@ -5172,6 +5182,33 @@ instruct storeNKlass(iRegN src, memory mem)
   ins_pipe(istore_reg_mem);
 %}
 
+// ============================================================================
+// Prefetch instructions
+// Must be safe to execute with invalid address (cannot fault).
+
+instruct prefetchalloc( memory mem ) %{
+  predicate(UseZicbop);
+  match(PrefetchAllocation mem);
+
+  ins_cost(ALU_COST * 3);
+  format %{ "prefetch_w $mem\t# Prefetch for write" %}
+
+  ins_encode %{
+    if ((($mem$$disp) & 0x1f) == 0) {
```

Comment:
Does this branch only check the alignment, not the allowed range?

Reply:
Fixed.